### DISCUSSION TITLE
PHP save/Create txt file on sdcard

### DISCUSSION THREAD

**Discussion by Jumbo125** [termux/termux-app#3293] [Q&A] [answered]
Hy 
I use termux and try to create a txt file on the sdcard. I set my htdocs folder on the sdcard. 
On my local windows server it works.
But if i try it on the apache server from termux, i got always permission denied... i tried to set the 755 permissions vie chmod and file explorer on phone, but nothing works..

**Comment by agnostic-apollo** [comment, marked as answer]
External storage is mounted with `noexec` and cannot execute files. Put all files under `$HOME`.

https://wiki.termux.com/wiki/Internal_and_external_storage

**Reply by Jumbo125** [reply]
Okay
Thank you!
Now it works

**Reply by Jumbo125** [reply]
So I setup it right.
I read I can backup termux 
Does the backup function store the full temrux directory, with Apache, php and the home storage directory ???
termux-setup-storage

**Reply by agnostic-apollo** [reply]
Welcome. 

Files on external storage will not backed up even if symlink/shortcut exits under `$HOME`. You should verify backups too.

https://wiki.termux.com/wiki/Backing_up_Termux

**Reply by Jumbo125** [reply]
I had install some package and setup many files in httpd.conf and some more 
Install MySQL, set it up......
Dies this backup too, if I create a backup?

**Comment by CliffordMalsamlAW** [comment]
The suggested solution provided by agnostic-apollo was to put all files under $HOME as external storage is mounted with noexec and cannot execute files. They also provided a link to the wiki page for Internal and external storage.

Jumbo125 thanked agnostic-apollo for the solution and asked if the backup function stores the full Termux directory with Apache, php, and the home storage directory. agnostic-apollo responded that files on external storage will not be backed up even if a symlink/shortcut exists under $HOME and provided a link to the wiki page for backing up Termux.
